framework/contracts/native/module-factory/src/commands.rs

```diff
@@ -89,12 +89,25 @@ pub fn execute_create_modules(
 
         match &new_module.reference {
             ModuleReference::App(code_id) => {
+                let init_msg = owner_init_msg.unwrap();
+                // App base message
+                let app_base_msg = abstract_core::app::BaseInstantiateMsg {
+                    ans_host_address: config.ans_host_address.to_string(),
+                    version_control_address: version_control.address.to_string(),
+                    account_base: account_base.clone(),
+                };
+
+                // TODO: App will have to do one extra deserialize, is it avoidable?
```

Comment:
@CyberHoward left this TODO, in case we find better solution

Reply:
You can use `Value`! https://crates.io/crates/serde-cw-value/0.7.0

Reply:
I'd prefer keeping as much type information as possible and working with Value where required.

Reply:
Yeah we still do one extra deserialize, but in module factory, so it's cleaner, thanks!

Reply:
You mean we deserialize the custom message into `Value` to then serialize it again within the bigger struct?